vars/cloverReportPublish.groovy

```diff
@@ -0,0 +1,128 @@
+// vars/cloverReportPublish.groovy
+
+  /**
+   * clover Report Publish step method
+   *
+   * Consolidiate clover data for publising.
+   *
+   * @param config Map of parameters passed
+   *
+   *
+   * config['coverage_healthy']    Map of a coverage healthy target.
+   *                               Default is:
+   *                                 [method:70, conditional:80, statement:80]
+   *
+   * config['coverage_script']     Script to run to get coverage results
+   *                               Default is "ci/bullseye_generate_report.sh"
+   *
+   * config['coverage_stashes']    list of stashes for coverage reports.
+   *                               Required.  Each stash must contain one
+   *                               file named test.cov.
+   *
+   * Config['coverage_website']    Zip file to contain the resulting
+   *                               coverage website, if it is a coverage
+   *                               build.  Defaults to 'coverage_website.zip'
+   *
+   * config['ignore_failure']      Ignore test failures.  Default false.
+   *
+   *
+   * config['inst_rpms']           Additional rpms to install.  Optional
+   *
+   * config['NODELIST']            NODELIST of nodes to run tests on.
+   *                               Default env.NODELIST
+   *
+   * config['node_count']          Count of nodes that will actually be used
+   *                               the test.  Default will be based on the
+   *                               enviroment variables for the stage.
+   *
+   * config['stash']               Stash name for the ".build-vars.*" files.
+   */
+
+def call(Map config = [:]) {
+
+  Map stage_info = parseStageInfo(config)
+
+  String inst_rpms = config.get('inst_rpms','')
+
+  if (stage_info['target'].startsWith("centos")) {
+    inst_rpms += ' java-1.8.0-openjdk'
+  } else if (stage_info['target'].startsWith('ubuntu')) {
+    inst_rpms += ' openjdk-8-jdk'
+  } else {
+    // Assume a Suse distro
+    inst_rpms += ' java-1_8_0-openjdk'
+  }
+
+  provisionNodes NODELIST: nodelist,
+                 node_count: stage_info['node_count'],
+                 distro: stage_info['target'],
+                 inst_repos: config.get('inst_repos',''),
+                 inst_rpms: inst_rpms
+
+  String coverage_website = config.get('coverage_website',
+                                       'coverage_website.zip')
+
+  String url_base = env.JENKINS_URL +
+                    'job/daos-stack/job/tools/job/master' +
+                    '/lastSuccessfulBuild/artifact/'
+
+  httpRequest url:  url_base + 'bullseyecoverage-linux.tar',
```

Comment:
Is this tarball available outside of our jenkins or do we build it in some way?  If we are just caching it, we should start to add branching here for getting things from their canonical source if the site doesn't have a local cache.